nexus/src/app/sagas/instance_create.rs

```diff
@@ -291,6 +308,77 @@ impl NexusSaga for SagaInstanceCreate {
     }
 }
 
+async fn sic_associate_ssh_keys(
+    sagactx: NexusActionContext,
+) -> Result<(), ActionError> {
+    let osagactx = sagactx.user_data();
+    let datastore = osagactx.datastore();
+    let saga_params = sagactx.saga_params::<Params>()?;
+
+    let opctx = crate::context::op_context_for_saga_action(
+        &sagactx,
+        &saga_params.serialized_authn,
+    );
+    let instance_id = sagactx.lookup::<Uuid>("instance_id")?;
+
+    // Gather the SSH public keys of the actor make the request so
+    // that they may be injected into the new image via cloud-init.
+    // TODO-security: this should be replaced with a lookup based on
```

Comment:
As above. Can you open an issue to track this and reference it here so this isn't forgotten about?

Reply:
I didn't add this TODO, just copied it over from where it was originally. I don't actually know what the original intent is here and I somewhat assume it's out of date. @davepacheco, I think you originally wrote this. Can you give any more context? I'm happy to file an issue, but I'd just like to understand more what the replacement should be (if it's even still relevant)

Reply:
I'm not sure.  Looking through the history I believe this came in with #974.  Maybe @plotnick remembers?

Reply:
Zowie, that was a while ago; thanks, @davepacheco for digging up that PR (#974). That TODO is mentioned in its description:

> Right now, we collect only the keys belonging to the actor making the request. Once role assignments for Silo users are in place, we can ask for, e.g., all the keys belonging to all the users with an appropriate role (vms.instance.create, probably) on the instance.

I don't remember exactly where that desire came from, other than generally wanting to allow customers to assign permissions at the role/group level rather than for individual users (as the current lookup would require); but we didn't have any kind of roles at the time, so couldn't do it then. If it's still desirable and we now have something like Silo/IdP roles (I've been away from that layer of Omicron for a while, so don't really know), it could certainly be spun out into an issue.